src/Analyser/Error.php

```diff
@@ -20,11 +20,17 @@ class Error
 	 */
 	private $line;
 
-	public function __construct(string $message, string $file, int $line = null)
+	/**
+	 * @var string|null
+	 */
+	private $rule;
```

Comment:
https://github.com/phpstan/phpstan/blob/f9c7a5674c83d43dfc41188062a229d75bddd57a/src/Analyser/Analyser.php#L174 and other places. I would like to see it nonnullable as well, any suggestions?